File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/EventNotificationSample/src/com/hp/workpath/sample/eventnotificationsample/task/AccessInitializationTask.java
package com.hp.workpath.sample.eventnotificationsample.task;

import android.content.Context;

import com.hp.workpath.api.access.AccessService;
import com.hp.workpath.sample.eventnotificationsample.R;

public class AccessInitializationTask extends InitializationTask {
    public AccessInitializationTask(Context context, ResultHandler handler) {
        super(context, handler);
    }

    @Override
    public boolean isSupported(Context context) {
        return AccessService.isSupported(context);
    }

    @Override
    public int getExceptionMessage() {
        return R.string.access_service_not_supported;
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/EventNotificationSample/src/com/hp/workpath/sample/eventnotificationsample/task/AccessInitializationTask.kt
package com.hp.workpath.sample.eventnotificationsample.task

import android.content.Context
import com.hp.workpath.api.access.AccessService
import com.hp.workpath.sample.eventnotificationsample.R

class AccessInitializationTask(context: Context, handler: ResultHandler) :
    InitializationTask(context, handler) {
    override fun isSupported(context: Context?): Boolean {
        return AccessService.isSupported(context?: return false)
    }

    override fun getExceptionMessage(): Int {
        return R.string.access_service_not_supported
    }
}